elastic/kibana
# [Dashboard] [Controls] Fix race condition on reset

## Summary

This PR fixes a race condition where new selections made in an options list control were not always reset as expected due to multiple input changes firing at the same time - because we were using separate subscriptions for these two input changes, their timing was not predictable, which meant that you could sometimes get unexpected results. 

This PR does two things to make this race condition less likely: 
1) it simplifies the logic for how the control group input is reset, which makes the timing of the input changes more predictable
2) it only resets the `timeRange` when necessary, which prevents the dashboard's `dashboardRefetchDiff` input subscription from firing on reset when it shouldn't (i.e. when `timeRestore` is `false`)

Refer to https://github.com/elastic/kibana/pull/156209#discussion_r1181730783 for a more verbose description of these changes + why they help fix this race condition.

### Before


https://user-images.githubusercontent.com/8698078/235259648-0f0923ff-9ad8-47b6-b814-5b036f20c80e.mov


Steps to recreate the bug:
1. Create an empty dashboard with at least one options list control (note that the dashboard doesn't necessarily have to be empty, but it is easier to consistently reproduce this bug with a fast loading dashboard).
2. Make some selections in the control and save the dashboard.
     <p align="center"><img src="https://user-images.githubusercontent.com/8698078/235253087-e4a49695-e547-4f7a-b9aa-d28c211cfefb.png"/></p>
3. Refresh the page (just to make sure any options list caching is cleared)
4. Make some new selections in the control, leaving the dashboard in an unsaved state.
     <p align="center"><img src="https://user-images.githubusercontent.com/8698078/235253396-b5cfc321-272f-4635-868d-a832ca97f337.png"/></p>
5. Reset the changes via the new "Reset" button.
6. You should see the control button text briefly be reset to the old "saved" selections from step 2, before it switches back to the "new" selections from step 3 - the `unsaved changes` badge will be gone, and the selections shown in the popover will be inconsistent with those shown in the button text:
     <p align="center"><img width="500px" src="https://user-images.githubusercontent.com/8698078/235252671-b8f78aae-6268-467b-9dd0-336eb55b29c2.png"/></p>

### After

https://user-images.githubusercontent.com/8698078/235259659-33712061-9d48-400c-939e-fcf2f2a1bfb7.mov

Steps to test:
1. Follow steps 1 through 5 in the above "Steps to recreate the bug" section
2. When the dashboard is reset, you should actually see the selections reset as expected - try this out with a few different dashboards to ensure that you don't see the weird "flicker" anymore

### Checklist



### For maintainers
